### PR TITLE
Include clangd-indexer in automatic builds

### PR DESCRIPTION
**Pull request by kirillbobyrev**

### CODE REVIEW

File: .github/workflows/autobuild.yaml [hunk 1]
```diff
@@ -120,12 +120,13 @@ jobs:
...
         ${{ matrix.config.cmake }}
     - name: Ninja
-      run: ninja -C ${{ env.CLANGD_DIR }} clangd
+      run: ninja -C ${{ env.CLANGD_DIR }} clangd clangd-indexer
     - name: Archive
       run: >
         7z a clangd.zip
         ${{ env.CLANGD_DIR }}/LICENSE.TXT
         ${{ env.CLANGD_DIR }}/bin/clangd*
+        ${{ env.CLANGD_DIR }}/bin/clangd-indexer
```
Comment (kadircet) [inline .github/workflows/autobuild.yaml:129]: drive-by comment:

I believe there would be value in patching those separately. as most of the users is likely to need clangd but not the indexer, and the additional binary would just increase package size.

(also you would want to put a `*` for matchin `.exe` on windows.

Comment (kirillbobyrev) [reply .github/workflows/autobuild.yaml]: Good point, thanks!

**Comment by kirillbobyrev** [comment]
@sam-mccall ping

**Comment by sam-mccall** [comment]
Sorry about the delay... I don't think this is a trivial question. It ties in to packaging, obviously.

I think the eventual state will be to have two packages, let's say `clangd` (current package, with remote index support) and `clangd-index-tools` (includes clangd-index-server and clangd-indexer):
 - the client and indexer parts have different target audiences, one much larger than the other
 - the client and server have a backwards-compatible interface, we can make them work across major versions. (Whereas the server and indexer share our binary format and basically need version lock)
This probably corresponds to basically the current version of the patch (after Kadir's suggestions), though it should also include clangd-index-server in the indexer package.

**However...**
For now, things are different:
 - `clangd` doesn't have the remote client on by default (and may not for a while)
 - `clangd`+grpc, `clangd-indexer` and `clangd-index-server` basically all have the same set of users (people experimenting with indexing)
 - the three are probably all version-locked in practice, as features are being added and we're not yet careful about wire-compatibility
 - the CI itself is likely to be unstable, need tweaking, not work on all platforms initially etc
So I think we want *one* package with matching version of clangd+grpc, clangd-indexer, and clangd-index-server. And for various reasons (single stream of releases per repo, confusion between versions of clangd, experimental CI churn) I think we should do this in a *different* repo.

So my suggestion would be:
 1. fork this repo (as `index-tools` unless you can suggest a better name), put a big "alpha" disclaimer on the fork
 2. modify the CI in the fork to enable grpc, include clangd-indexer and clangd-index-server, and to rename the zip files for clarity (in any order)
 3. no code review required on the fork (happy to if you want it though)
 4. once this has stabilized, merge it into the main repo, producing two zip files

WDYT?

Had an offline discussion and @kadircet is also on OK with this (though not as convinced as I am about the benefit of a new repo).

**Comment by kirillbobyrev** [comment]
That makes sense, thank you for the idea!
